staker-rewards/lib/rewards-calculator.js

```diff
@@ -98,6 +107,45 @@ export default class RewardsCalculator {
     return new BigNumber(minimumStake)
   }
 
+  checkRequirementsViolations(operatorParameters) {
+    const violations = []
+    const { isFraudulent, requirements, operatorSLA } = operatorParameters
+
+    if (isFraudulent === true) {
+      violations.push("isFraudulent")
+    }
+
+    if (requirements.factoryAuthorizedAtStart === false) {
+      violations.push("factoryAuthorizedAtStart")
+    }
+
+    if (requirements.poolAuthorizedAtStart === false) {
+      violations.push("poolAuthorizedAtStart")
+    }
+
+    if (requirements.poolDeauthorizedInInterval === true) {
+      violations.push("poolDeauthorizedInInterval")
+    }
+
+    if (requirements.minimumStakeAtStart === false) {
+      violations.push("minimumStakeAtStart")
+    }
+
+    if (requirements.poolRequirementFulfilledAtStart === false) {
+      violations.push("poolRequirementFulfilledAtStart")
+    }
+
+    if (operatorSLA.keygenSLA !== "N/A" && operatorSLA.keygenSLA < 90) {
+      violations.push("keygenSLA")
+    }
+
+    if (operatorSLA.signatureSLA !== "N/A" && operatorSLA.signatureSLA < 95) {
+      violations.push("signatureSLA")
+    }
```

Comment:
The rules we discussed are:

If the operator does not meet signature SLA between T_start and T_end , they are not getting any rewards for interval T . SLA is:
- For 20 or less closed/terminated keeps, no more than one with a signature timeout.
- For more than 20 closed/terminated keeps/ no more than 5% with a signature timeout.

Example:
For less than 20 redemptions, one of them can fail.
For 20-40 redemptions, only 1 can fail.  
For 50 redemptions, only 2 can fail.

(Signature SLA does NOT include not answered courtesy calls.)

If the operator has more than 10% of unsuccessful key generations between T_start and T_end , they are not getting any rewards for interval T. SLA is:
- For 2 or less key generations, none of them can fail.
- For 2-10 or less key generations, one of them can fail.
- For more than 10 key generations, no more than 10% can fail.

Example:
For 2 key generations, none of them can fail.
For 10 key generations, 1 of them can fail.
For 20 key generations, 2 of them can fail.
For 50 key generations, 5 of them can fail.

Let's also include a CAPITAL LETTER social caveat in the code that if we see DoS behaviors in the network, we can adjust SLAs.

Reply:
Ok, done.